### PR TITLE
Integrate gutenberg-mobile release 1.99.0

### PR DESCRIPTION
**Pull request by fluiddot**
## Description
This PR incorporates the 1.99.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5944

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -3,6 +3,8 @@
 22.8
 -----
 * [*] Fixes the crash that occurs after deleting the site [https://github.com/wordpress-mobile/WordPress-Android/pull/18736]
+* [*] [Jetpack-only] Block editor: Rename "Reusable blocks" to "Synced patterns", aligning with the web editor. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/5885]
+* [**] Block editor: Fix a crash related to Reanimated when closing the editor [https://github.com/wordpress-mobile/gutenberg-mobile/pull/5938]
```
Comment (geriux) [inline RELEASE-NOTES.txt]: I wonder if we should mark this as internal note 🤔 what do you think?

**Review by geriux — APPROVED** [review]
LGTM! Friendly reminder to update the Gutenberg mobile commit hash to the release tag. 🚀

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (fluiddot) [reply RELEASE-NOTES.txt]: Yep, good idea. Although it's a crash (i.e. something the user notices), I think the release note doesn't give any value to the user as it's hard to reproduce in the app. I'll update it before merging the PR. Thanks!

Comment (geriux) [reply RELEASE-NOTES.txt]: Maybe we could mention something when this is sent to editorial, to highlight that this release includes a fix for random crashes when closing the editor.

Comment (fluiddot) [reply RELEASE-NOTES.txt]: Yeah although thinking it twice, maybe we should wait until we get the first crash metrics, once it's released, to confirm that the random crashes are fixed. I'm concerned that the fix doesn't improve those random crashes.

Comment (geriux) [reply RELEASE-NOTES.txt]: True, yeah let's do that later on once we confirm it is fixed 👍

**Pull request merged**